app/controllers/images_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ImagesController < ApplicationController
+
+  before_action :authenticate_user!, except: [:index, :show]
+
+  def index
+    @images = Image.order(likes: :desc)
+  end
+
+  def new
+    @image = Image.new
+  end
+
+  def create
+    @image = current_user.images.new(image_params)
+    if @image.save
+      redirect_to images_path
+    else
+      flash[:notice] = @image.errors.full_messages.join(', ')
+      redirect_to new_image_path
+    end
+  end
+
+  def show
+    @image = Image.find(params[:id])
```

Comment:
You've got the line `@image = Image.find(params[:id])` three times. Can you think of a way to refactor?